### PR TITLE
More words and explain -release

### LINKED ISSUE #2308: Explain why to use -target -release etc
## reproduction steps

It comes up all the time. Is there a plugin I can use to automate my expectations? `sbt-seth`?

https://discord.com/channels/632150470000902164/632150470000902166/938864015545880626

## problem

https://docs.scala-lang.org/overviews/jdk-compatibility/overview.html should probably explain option selection.

### PR DESCRIPTION
**Pull request by som-snytt**
Fixes #2308

### CODE REVIEW

File: _overviews/compiler-options/index.md [hunk 1]
```diff
@@ -25,17 +25,13 @@ title: Scala Compiler Options
 
 ## Introduction
 
-Scala compiler `scalac` offers various **compiler options**, also referred to as **compiler flags**, to change how to compile your program.
+The Scala compiler `scalac` offers various **compiler options**, or **flags**, that change the compiler's default behavior. Some options just generate more compiler output in the form of diagnostics or warnings, while others change the result of compilation.
 
-Nowadays, most people are not running `scalac` from the command line.
-Instead, they use sbt, an IDE, and other tools as their interface to the compiler.
-Therefore they may not even have `scalac` installed, and won't think to do `man scalac`.
+The Scala command `scala`, which runs scripts or compiled code, accepts the compiler options plus a few more that determine how to run a program.
```
Comment (bjornregnell) [inline _overviews/compiler-options/index.md:30]: perhaps change `the compiler options` to `all compiler options` ?

Comment (bjornregnell) [reply _overviews/compiler-options/index.md:30]: or `most compiler options` if not exactly all are accepted?

File: _overviews/compiler-options/index.md [hunk 2]
```diff
@@ -44,34 +40,41 @@ This page comes to the rescue for the people to find&hellip;
...
-E.g. `scalac -encoding utf8 -Xfatal-warnings Hello.scala`
+Options that take just a single argument accept traditional syntax:
 
-Default paths can be listed by running a command line tool:
+`scalac -d /tmp`
+
+Default paths can be listed by running a tool in the distribution:
 ```
 scala scala.tools.util.PathResolver [ <options> ]
 ```
```
Comment (bjornregnell) [inline _overviews/compiler-options/index.md:59]: This does not work for me:
```
$ scala scala.tools.util.PathResolver
Ignoring spurious arguments: scala.tools.util.PathResolver
Welcome to Scala 3.1.3 (17.0.3, Java OpenJDK 64-Bit Server VM).
Type in expressions for evaluation. Or try :help.
                                                                                                                               
scala> 
```

Comment (bjornregnell) [reply _overviews/compiler-options/index.md:59]: Am I doing something wrong?

File: _overviews/compiler-options/index.md [hunk 3]
```diff
@@ -44,34 +40,41 @@ This page comes to the rescue for the people to find&hellip;
...
+Here is a typical configuration of the `scalacOptions` setting in `sbt`:
 
 ```scala
 scalacOptions ++= Seq(
-  "-encoding", "utf8", // Option and arguments on same line
-  "-Xfatal-warnings",  // New lines for each options
-  "-deprecation",
-  "-unchecked",
+  "-encoding", "utf8",          // if an option takes an arg, supply it on the same line
+  "-feature",                   // put the next option on a new line
```
Comment (bjornregnell) [inline _overviews/compiler-options/index.md:67]: The newline is just for readability and not illegal? Perhaps change the comment then to:
`// for better readability, put the next option on a new line`

**Review by bjornregnell — CHANGES_REQUESTED** [review]
Looks good to me, but you may want to include my suggestions (esp. the error on PathResolver that does not seem to work for Scala 3 should be addressed somehow, if it is not just me who is getting "Ignoring spurious arguments:" )

File: _overviews/compiler-options/index.md (hunk 1 above)
Comment (som-snytt) [reply _overviews/compiler-options/index.md:30]: "the same options as the scalac compiler", where "the compiler options" meant "options to the compiler".

File: _overviews/compiler-options/index.md (hunk 2 above)
Comment (som-snytt) [reply _overviews/compiler-options/index.md:59]: It's scala 2, `scala.tools` vs `dotty.tools`. Worth mentioning.

File: _overviews/compiler-options/index.md (hunk 3 above)
Comment (som-snytt) [reply _overviews/compiler-options/index.md:67]: it improves writability, so I said something like that. Autocorrect says "irritability", which is closer to the truth.

**Comment by som-snytt** [comment]
Thanks @bjornregnell

**Review by bjornregnell — APPROVED** [review]

**Pull request merged**